WallsAndHoles/tiletemplate.h

```diff
@@ -67,6 +106,27 @@ class TileTemplate : public QObject
     //The color the tile will be in the map view.
     //Has no affect on evental output mesh
     QColor mColor;
+
+    /**
+     * @brief The material this tile template uses.
+     *
+     * For now, all tile templates will use SimpleTexturedMaterials. Later on,
+     * when we choose to add in cooler rendering, SimpleTexturedMaterial
+     * may get extended.
+     *
+     * This is a pointer to allow for polymorphism.
+     * This is a smart pointer to avoid memory leaks and to make ownership explicit.
+     * This is a unique pointer because the material should only be editable from within this class.
+     *
+     * Implementation note: TileTemplate should never construct a SimpleTexturedMaterial
+     * because it should not assume what subclass of the material is being used.
+     */
```

Comment:
So this is the main design point we should talk about. A Template, in my mind, should never own a Material, so a sharedPointer should probably be used instead, maybe even a pointer to a const material for that matter. We will discuss this design further